Code cleanup

go4retro opened the pull request:
- define enums for magic numbers in the code

- optimized return_* functions to remove duplicate code, and fix some potential issues in original code

go4retro committed:
Made a few mods to bring the SW up on Arduino UNO.  Still not functional

go4retro committed:
Add in .gitignore file

go4retro committed:
Prep for merge

go4retro committed:
Merge branch 'master' of https://github.com/bkw777/PDDuino

go4retro committed:
Turn off debug, don't send hidden files, a bit more debugging

go4retro committed:
Small alignment changes

go4retro committed:
Merge branch 'master' of https://github.com/bkw777/PDDuino into master and add support for MEGA2560

go4retro committed:
Create command_t enum

go4retro committed:
Refactored return_* functions and fixed a bug withen.  code size down from 21740->21660

go4retro committed:
Getting rid of MAGIC NUMBERS in the code

bkw777 commented:
I'm good with most of this but a lot of these things need to be split into their own commits and tested in test branches.

Right now the main code mostly works and I actively test it on 3 platforms (4 boards, but the two teensy's take all the same code) and try to make sure that it doesn't break or get worse before actually merging into master, even though master is hardly in great shape itself yet. I don't want merge this whole thing like it is right now into master. But for instance, I'd do the enums in a enum branch, and then after I test it on the existing real hardware, maybe work on it a little more if needed, then marge that into master. Same for adding the new platforms.

I think it's also time to move things into separate files, like all the platform-specific stuff at the top into a platforms.h or maybe even a small .h for each platform, even though it would just be a few lines in each case, only the lines that differed from the template.

bkw777 commented:
This is awesome btw and thank you for working on it.

go4retro committed:
Removed overly complicated ring buffer and added a few more enumerations

go4retro committed:
Added in System State functionality, to guard against using command at the wrong time, and created some enumerations to remove magic numbers in the code.

go4retro committed:
Add in some more command definitions and implement command checksum validation

go4retro committed:
General cleanup and addition of more extended commands

* Better handle erroneous incoming data like ZM1ZZ and such
* Add in more extended commands
* Support > 65535 byte files in Reference return

go4retro committed:
SWitched to cleaner Logging facility and started to break apart code

go4retro committed:
More refacting (tpdd functions) and fix timing issue

* Refactor tpdd functions into separate file
* refactor led defines to save code space
* Reduce logging amount so DME request doesn't time out

go4retro committed:
Refactor #defines and split out power mgmt functions

* Switch LED defines to static inline functions, as that saves FLASH
* moved SLEEP functions to powermgmt.cpp/h
* cleaned up SendLoader progress bar.

go4retro committed:
More refactoring and cleanup

* refactored DTR and DSR pins to save code and minimize #ifdefs
* conditional compilation of TPDD Extensions
* create LOG_INT() function to handle starting LOG
* Eclipse messed up enum formatting... Fixed.

go4retro committed:
Create proper src dir for potential Makefile use

go4retro committed:
Add in proper GPLv3 flowerboxes and add in Logger.h to src dir

go4retro committed:
Create board_init() function for each baord options

go4retro committed:
Merge in commit: ef4831fdd2380a3d7a1e2e8bb13f12b2626a89db

https://github.com/bkw777/PDDuino/commit/ef4831fdd2380a3d7a1e2e8bb13f12b26
Work-around Adafruit broke their SAMD21 support
github.com/adafruit/ArduinoCore-samd/issues/20926a89db

go4retro committed:
Merge in commit: 4953717c6517168f834d6f34c9d3c7b967cb16b7

https://github.com/bkw777/PDDuino/commit/4953717c6517168f834d6f34c9d3c7b967cb16b7

go4retro committed:
Merge in the rest of commit 4953717c6517168f834d6f34c9d3c7b967cb16b7

https://github.com/bkw777/PDDuino/commit/4953717c6517168f834d6f34c9d3c7b967cb16b7

go4retro committed:
Update README.md (commit: 3b73c27932a51087cc76312fd39a9b5cbfc8bfaa)

https://github.com/bkw777/PDDuino/commit/3b73c27932a51087cc76312fd39a9b5cbfc8bfaa

go4retro committed:
Update readme more (Commit: b5f3f8b4eadf7b4585ce9c98ade1687727a3a0fb)

https://github.com/bkw777/PDDuino/commit/b5f3f8b4eadf7b4585ce9c98ade1687727a3a0fb

go4retro committed:
Merge in commit: 4a52e96fcb38bcee25312c357268522b04ec5a9a

https://github.com/bkw777/PDDuino/commit/4a52e96fcb38bcee25312c357268522b04ec5a9a

go4retro committed:
Merge in commit: dd89f8e1ea3896381a32e922e423dc5be10c4ee9

https://github.com/bkw777/PDDuino/commit/dd89f8e1ea3896381a32e922e423dc5be10c4ee9
